### PR TITLE
configure: Some minor fixes and improvements

### PR DESCRIPTION
**Pull request by mobin-2008**
Hello. Changes:

1. `sub_warning()` is combined with `warning()` for better usage
2. Some new short options (`-h`, `-q`, `-c`)
3. Fixes "TEST_CXXFLAGS_EXTRA" which got ignored
4. Fixes unnecessary forking for writing into mconfig
5. `--clean` renamed to `--clear` (I think it's better now)

`Signed-off-by: Mobin "Hojjat" Aydinfar <mobin@mobintestserver.ir>`

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -39,16 +39,13 @@ sub_error()
     exit 1
 }
 
+# NOTE: Always quote your warning() messages
 warning()
```
Comment (davmac314) [inline configure:43]: Add a comment explaining what parameters can/must be supplied (i.e. warning message and optional extra information)

File: configure [hunk 2]
```diff
@@ -39,16 +39,13 @@ sub_error()
...
-    echo "Warning: $*"
-}
-
-sub_warning()
-{
-    echo "  ... $*"
-    echo
+    >&2 echo
+    >&2 echo "Warning: $1"
+    [ -n "${2:-}" ] && >&2 echo " ... $2" # Built-in sub_warning()
```
Comment (davmac314) [inline configure:47]: The "built-in sub_warning()" comment doesn't make sense to someone reading the code, because sub_warning doesn't exist. You could probably remove the comment altogether (once you've documented the parameters, see comment above).

File: configure [hunk 3]
```diff
@@ -39,16 +39,13 @@ sub_error()
...
+    >&2 echo
 }
```
Comment (davmac314) [inline configure:49]: Why combine `warning` with `sub_warning` but not `error` with `sub_error`? This is inconsistent.

File: configure [hunk 4]
```diff
@@ -228,7 +224,7 @@ for arg in "$@"; do
         --disable-auto-restart|--enable-auto-restart=no) DEFAULT_AUTO_RESTART=false ;;
         CXX=*|CXX_FOR_BUILD=*|CXXFLAGS_FOR_BUILD=*|CPPFLAGS_FOR_BUILD=*\
         |LDFLAGS_FOR_BUILD=*|CXXFLAGS=*|CXXFLAGS_EXTRA=*|TEST_CXXFLAGS=*\
-        |TEST_CXXFLAGS_EXTRA|LDFLAGS=*|LDFLAGS_EXTRA=*|TEST_LDFLAGS=*\
+        |TEST_CXXFLAGS_EXTRA=*|LDFLAGS=*|LDFLAGS_EXTRA=*|TEST_LDFLAGS=*\
```
Comment (davmac314) [inline configure:227]: Please separate fixes into separate commits from other refactoring/rework

**Review by davmac314 — CHANGES_REQUESTED** [review]

File: configure (hunk 3 above)
Comment (mobin-2008) [reply configure:49]: Mainly because you can use sub_error() with info() without error() but warning() (and sub_warning()) didn't used like this.

Comment (davmac314) [reply configure:51]: This seems borderline to me, but ok.

**Review by davmac314 — APPROVED** [review]

File: configure [hunk 5]
```diff
@@ -391,11 +388,11 @@ if [ -n "${USE_UTMPX:-}" ]; then
     echo "USE_UTMPX=$USE_UTMPX" >> mconfig
 fi
 if [ -n "${CXX_FOR_BUILD:-}" ]; then
-    {
+    (
```
Comment (davmac314) [inline configure:391]: This is wrong, it spawns a new subshell. The original code was correct.
